### PR TITLE
Explicitly state version for removal of deprecated options

### PR DESCRIPTION
**Pull request by sushantmimani**
To help us get this pull request reviewed and merged quickly, please be sure to include the following items:

* [ ] Tests (if applicable)
* [ ] Documentation (if applicable)
* [ ] Changelog entry
* [x] A full explanation here in the PR description of the work done

Update deprecation warnings to state that all deprecated options will be remove from Tartufo 4.x

## PR Type
What kind of change does this PR introduce?

* [ ] Bugfix
* [ ] Feature
* [ ] Code style update (formatting, local variables)
* [ ] Refactoring (no functional changes, no api changes)
* [ ] Build related changes
* [ ] CI related changes
* [ ] Documentation content changes
* [ ] Tests
* [x] Other

## Backward Compatibility

Is this change backward compatible with the most recently released version? Does it introduce changes which might change the user experience in any way? Does it alter the API in any way?

* [x] Yes (backward compatible)
* [ ] No (breaking changes)

### CODE REVIEW

File: tartufo/config.py [hunk 1]
```diff
@@ -188,8 +188,8 @@ def configure_regexes(
 
     if rules_files:
         warnings.warn(
-            "Storing rules in a separate file has been deprecated and will be removed "
-            "in a future release. You should be using the 'rule-patterns' config "
+            "Storing rules in a separate file is deprecated and will be removed "
+            "in Tartufo 4.x. You should be using the 'rule-patterns' config "
```
Comment (rbailey-godaddy) [inline tartufo/config.py]:
```suggestion
            "in tartufo 4.x. Please use the 'rule-patterns' config "
```
I checked usage in tartufo.readthedocs.io and the canonical name appears to be all lowercase.

File: tartufo/scanner.py [hunk 2]
```diff
@@ -178,7 +178,8 @@ def hex_entropy_limit(self) -> float:
         # # directly (but complain about it).
         if self.global_options.hex_entropy_score:
             warnings.warn(
-                "--hex-entropy-score is deprecated. Use --entropy-sensitivity instead.",
+                "--hex-entropy-score is deprecated and will be removed in Tartufo 4.x. "
+                "Use --entropy-sensitivity instead.",
```
Comment (rbailey-godaddy) [inline tartufo/scanner.py]:
```suggestion
                "--hex-entropy-score is deprecated and will be removed in tartufo 4.x. "
                "Please use --entropy-sensitivity instead.",
```

File: tartufo/scanner.py [hunk 3]
```diff
@@ -196,7 +197,8 @@ def b64_entropy_limit(self) -> float:
         # # directly (but complain about it).
         if self.global_options.b64_entropy_score:
             warnings.warn(
-                "--b64-entropy-score is deprecated. Use --entropy-sensitivity instead.",
+                "--b64-entropy-score is deprecated and will be removed in Tartufo 4.x. "
+                "Use --entropy-sensitivity instead.",
```
Comment (rbailey-godaddy) [inline tartufo/scanner.py]:
```suggestion
                "--b64-entropy-score is deprecated and will be removed in tartufo 4.x. "
                "Please use --entropy-sensitivity instead.",
```

File: tartufo/scanner.py [hunk 4]
```diff
@@ -267,7 +269,7 @@ def included_paths(self) -> List[Pattern]:
             if deprecated:
                 warnings.warn(
                     "Old format of --include-path-patterns option and config file setup include-path-patterns "
-                    "= ['inclusion pattern'] has been deprecated and will be removed in a future version. "
+                    "= ['inclusion pattern'] has been deprecated and will be removed in Tartufo 4.x. "
```
Comment (rbailey-godaddy) [inline tartufo/scanner.py]:
```suggestion
                    "= ['inclusion pattern'] has been deprecated and will be removed in tartufo 4.x. "
```

File: tartufo/scanner.py [hunk 5]
```diff
@@ -316,7 +318,7 @@ def excluded_paths(self) -> List[Pattern]:
             if deprecated:
                 warnings.warn(
                     "Old format of --exclude-path-patterns option and config file setup exclude-path-patterns "
-                    "= ['exclusion pattern'] has been deprecated and will be removed in a future version. "
+                    "= ['exclusion pattern'] has been deprecated and will be removed in Tartufo 4.x. "
```
Comment (rbailey-godaddy) [inline tartufo/scanner.py]:
```suggestion
                    "= ['exclusion pattern'] has been deprecated and will be removed in tartufo 4.x. "
```

File: tartufo/scanner.py [hunk 6]
```diff
@@ -404,7 +406,7 @@ def excluded_signatures(self) -> Tuple[str, ...]:
             if deprecated:
                 warnings.warn(
                     "Configuring exclude-signatures as string has been deprecated and support for this format will "
-                    "be removed in the future. Please make sure to update your exclude-signatures configuration to "
+                    "be removed in Tartufo 4.x. Please make sure to update your exclude-signatures configuration to "
```
Comment (rbailey-godaddy) [inline tartufo/scanner.py]:
```suggestion
                    "be removed in tartufo 4.x. Please update your exclude-signatures configuration to "
```

**Review by rbailey-godaddy — CHANGES_REQUESTED** [review]
Editorial wordsmithing.

**Review by rbailey-godaddy — APPROVED** [review]

**Review by emayuri-godaddy — APPROVED** [review]

**Pull request merged**